### PR TITLE
(Iris4) - Did some fix and optimization

### PR DESCRIPTION
**Pull request by XIAYM-gh**
Changes I've made:
 - Optimized some I/O related code (and others)
 - Fixed server type getting
 - Possibly fixed pack compiling

;) Thank you for reviewing.

### CODE REVIEW

File: core/src/main/java/com/volmit/iris/core/service/StudioSVC.java [hunk 1]
```diff
@@ -369,6 +369,7 @@ public void close() {
     }
 
     public File compilePackage(VolmitSender sender, String d, boolean obfuscate, boolean minify) {
+        System.out.println(new File(getWorkspaceFolder(), d) + " -> " + new File(getWorkspaceFolder(), d).exists());
```
Comment (CrazyDev05) [inline core/src/main/java/com/volmit/iris/core/service/StudioSVC.java]: don't use `System.out`, use Iris::info instead

Comment (XIAYM-gh) [reply core/src/main/java/com/volmit/iris/core/service/StudioSVC.java]: Whoa, my fault. This line is written for debugging, I forgot to remove it

File: core/src/main/java/com/volmit/iris/Iris.java [hunk 2]
```diff
@@ -899,8 +883,9 @@ public void splash() {
 
         setupChecks();
         Iris.info("Java: " + getJava());
-        if (!instance.getServer().getVersion().contains("Purpur")) {
-            if (instance.getServer().getVersion().contains("Spigot") && instance.getServer().getVersion().contains("Bukkit")) {
+        String serverName = instance.getServer().getName();
+        if (!serverName.contains("Purpur")) {
+            if (serverName.contains("Spigot") && serverName.contains("Bukkit")) {
```
Comment (CrazyDev05) [inline core/src/main/java/com/volmit/iris/Iris.java:888]: This need to be
```java
if (serverName.contains("Spigot") || serverName.contains("Bukkit")) {
```

**Comment by CrazyDev05** [comment]
Closed due to inactivity

**Pull request closed without merging**